src/GraphQL/Introspection/ISchemaFilter.cs

```diff
@@ -44,25 +44,48 @@ public interface ISchemaFilter
     }
 
     /// <summary>
-    /// The default schema filter. By default nothing is hidden.
+    /// The default schema filter. By default nothing is hidden. Please note that
+    /// some features that are not in the official specification may be hidden
+    /// by default. These features can be unlocked using your own filter.
     /// </summary>
     public class DefaultSchemaFilter : ISchemaFilter
     {
-        private static readonly Task<bool> _completed = Task.FromResult(true);
+        private static readonly Task<bool> _allowed = Task.FromResult(true);
+        private static readonly Task<bool> _forbidden = Task.FromResult(false);
 
         /// <inheritdoc/>
-        public virtual Task<bool> AllowType(IGraphType type) => _completed;
+        public virtual Task<bool> AllowType(IGraphType type) => type is __AppliedDirective || type is __DirectiveArgument ? _forbidden : _allowed;
```

Comment:
I think that hardcode the values ​​in this place is not the best way to hide the directive introspection feature. 
For someone unfamiliar with the implementation details, it will be unclear how to enable the feature. 
It seems to me that it would be better to move the enable / disable of this feature to the Schema level.

Something like 
```c#
var shcema = new Schema(); 
shcema.UseDirectiveIntrospection() 
``` 
would be great

Reply:
We certainly should well-document it. Maybe we could at least provide two schema filters even if the default hides the out-of-spec fields.

Reply:
Done:
```c#
 public static TSchema EnableExperimentalIntrospectionFeatures<TSchema>(this TSchema schema)
            where TSchema : Schema
        {
            schema.Filter = new ExperimentalFeaturesSchemaFilter();
            return schema;
        }
```

Reply:
Nice